pygeoapi/provider/postgresql.py

```diff
@@ -279,11 +281,15 @@ def create(self, item, crs_transform_func=None):
             for k, v in properties.items():
                 json_data[k] = v
         db_obj_mapping = self.table_model(**json_data)
-        with Session(self._engine) as session:
-            session.add(db_obj_mapping)
-            session.commit()
-            session.refresh(db_obj_mapping)
-            identifier = getattr(db_obj_mapping, self.id_field)
+        try:
+            with Session(self._engine) as session:
+                session.add(db_obj_mapping)
+                session.commit()
+                session.refresh(db_obj_mapping)
+                identifier = getattr(db_obj_mapping, self.id_field)
+        except DataError as err:
```

Comment:
Hi @francbartoli, just checked for valid GeoJSON before that point. Was not sure if instantiating `db_obj_mapping` with *wrong* data passed to the table model would throw an error. No error came until the `session.commit()` statement. Not sure what is the best way to implement it. I also think it would be better with a data validation layer using schemas. As for now, I do not think pygeoapi serve collections' schema, and in order to do so, we would need to implement a *schema generation* function for all providers. Right now, I am working on adding transactions support for `PostgreSQLProvider`, and I though I could let PostgreSQL/PostGIS throw an error if the `POST`-ed data is invalid and catch the exception. `ElasticsearchProvider` supports transactions but is not using schema validation.

Reply:
in this issue https://github.com/opengeospatial/ogcapi-features/issues/424 the `mutation schema` is mentioned for that purpose, @tomkralidis any clue why it is completely missing from Part-4 now?

Reply:
@francbartoli , I think you will find your answer there: https://github.com/opengeospatial/ogcapi-features/issues/740

Reply:
Thanks @MTachon, this is a WIP indeed. Can you please put a comment that this would be abstracted as soon as this part of the specification will be released and implemented in pygeoapi?

Reply:
> Hi @francbartoli, just checked for valid GeoJSON before that point. Was not sure if instantiating `db_obj_mapping` with _wrong_ data passed to the table model would throw an error. No error came until the `session.commit()` statement. Not sure how it should be implemented. I also think it would be better with a data validation layer using schemas. As for now, I do not think pygeoapi serves collections' schemas, and in order to do so, we would need to implement a _schema generation_ function for all providers. Right now, I am working on adding transactions support for `PostgreSQLProvider`, and I thought I could let PostgreSQL/PostGIS throw an error if the `POST`-ed data is invalid and catch the exception. `ElasticsearchProvider` supports transactions but is not using schemas validation.

@MTachon FYI schema generation was implemented in https://github.com/geopython/pygeoapi/pull/1022

Reply:
@tomkralidis , thanks for pointing it out! I was not aware of it, I will have a look.